ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -946,3 +891,416 @@ def test_ipa_hbac_rule_duplication(self):
+        """
+        This test checks that ipa-migrate is successful
+        in prod run mode
+        """
+        tasks.kinit_admin(self.master)
+        tasks.kinit_admin(self.replicas[0])
+        MIGRATION_SCHEMA_LOG_MSG = "Migrating schema ...\n"
+        MIGRATION_DATABASE_LOG_MSG = (
+            "Migrating database ... (this may take a while)\n"
+        )
+        IPA_UPGRADE_LOG_MSG = (
+            "Running ipa-server-upgrade ... (this may take a while)\n"
+        )
+        SIDGEN_TASK_LOG_MSG = "Running SIDGEN task ...\n"
+        result = run_migrate(
+            self.replicas[0],
+            "prod-mode",
+            self.master.hostname,
+            "cn=Directory Manager",
+            self.master.config.admin_password,
+            extra_args=['-n'],
+        )
+        install_msg = self.replicas[0].get_file_contents(
+            paths.IPA_MIGRATE_LOG, encoding="utf-8"
+        )
+        assert result.returncode == 0
+        assert MIGRATION_SCHEMA_LOG_MSG in install_msg
+        assert MIGRATION_DATABASE_LOG_MSG in install_msg
+        assert IPA_UPGRADE_LOG_MSG in install_msg
+        assert SIDGEN_TASK_LOG_MSG in install_msg
+
+    def test_ipa_migrate_prod_mode_hbac_rule(self, empty_log_file):
+        """
+        This testcase checks that hbac rule is migrated from
+        remote server to local server in prod mode.
+        """
+        hbac_rule_name1 = 'test1'
+        hbac_rule_name2 = 'testuser_sshd'
+        tasks.kinit_admin(self.replicas[0])
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "hbacrule-find", hbac_rule_name1])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "hbacrule-find", hbac_rule_name2])
+        assert hbac_rule_name1 in cmd1.stdout_text
+        assert hbac_rule_name2 in cmd2.stdout_text
+
+    def test_ipa_migrate_prod_mode_sudo_rule(self, empty_log_file):
+        """
+        This testcase checks that sudo cmd and rules are
+        migrated from remote server to local server in prod mode.
+        """
+        sudorule = 'readfiles'
+        sudocmd = '/usr/bin/less'
+        tasks.kinit_admin(self.replicas[0])
+        cmd1 = self.replicas[0].run_command(
+            ["ipa", "sudorule-find", sudorule])
+        cmd2 = self.replicas[0].run_command(
+            ["ipa", "sudocmd-find", sudocmd])
+        assert 'Rule name: readfiles\n' in cmd1.stdout_text
+        assert 'Sudo Command: /usr/bin/less\n' in cmd2.stdout_text
+
+    def test_ipa_migrate_prod_mode_new_user_sid(self):
+        """
+        This testcase checks that in prod-mode uid/gid of the
+        migrated user is preserved i.e we have same
+        uid/gid for user on remote and local IPA server.
+        """
+        username = 'produser1'
+        original_passwd = 'Secret123'
+        tasks.user_add(self.master, username, password=original_passwd)
+        run_migrate(
+            self.replicas[0],
+            "stage-mode",
```

Comment:
You can simplify a lot this test. The test class inherits from MigrationTest, meaning that the server is populated by the prepare_ipa_server method. ipa-migrate is already called in the test test_ipa_migrate_prod_mode.
The test only needs to call `id username ` with one of the users created in prepare_ipa_server, for instance testuser1.